sql/log_event_client.cc

```diff
@@ -648,7 +655,14 @@ log_event_print_value(IO_CACHE *file, PRINT_EVENT_INFO *print_event_info,
 
       float8get(dbl, ptr);
       char tmp[320];
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
       sprintf(tmp, "%-.20g", dbl); /* strmake doesn't support %-20g */
+#ifdef __clang__
```

Comment:
I think compilers could determine if this is safe with a sizeof rather than warning suppression.

Could expand on safe vs not safe in commit message.